### ISSUE TITLE
huawei v10 p20 can not trace

### ISSUE THREAD

**Issue by CatDroid** [google/gapid#2571]
I use GAPID on huawei v10 and p20 but it fail after startup,and many logs looks like 
```
01-30 10:47:22.816 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glPathGlyphIndexRangeNV at 0xe6836804
01-30 10:47:22.816 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glCopyTexSubImage3D at 0xe6835db4
01-30 10:47:22.817 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glGetMaterialxv at 0xe68346c8
01-30 10:47:22.817 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glReadnPixels at 0xe6836568
01-30 10:47:22.817 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glGenRenderbuffers at 0xe6835cb0
01-30 10:47:22.818 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glSampleCoveragex at 0xe6834ee4
01-30 10:47:22.818 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glRenderbufferStorageMultisampleNV at 0xe68364a0
01-30 10:47:22.818 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glPathMemoryGlyphIndexArrayNV at 0xe683680c
01-30 10:47:22.819 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glBufferStorageEXT at 0xe6836684
01-30 10:47:22.819 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glTexEnvxOES at 0xe683544c
01-30 10:47:22.819 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glDrawElementsBaseVertex at 0xe6836550
01-30 10:47:22.820 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glMultMatrixf at 0xe6834180
01-30 10:47:22.821 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glProgramUniform1ivEXT at 0xe68363a4
01-30 10:47:22.822 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glDebugMessageInsertKHR at 0xe683609c
```
how can i fix it ?

**Comment by CatDroid** [comment]
add more error log info 

```
01-30 10:47:22.347 21230 21230 E GAPID   : [gapii/cc/spy.cpp:178] APIS 0000000a
01-30 10:47:22.498 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glWeightPointerOESBounds at 0xe53325cc
01-30 10:47:22.499 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glViewportIndexedfvNV at 0xe6836830
01-30 10:47:22.499 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glViewportIndexedfNV at 0xe683682c
01-30 10:47:22.500 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glViewportArrayvNV at 0xe6836828
01-30 10:47:22.501 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexPointerBounds at 0xe533256c
01-30 10:47:22.502 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexPointer at 0xe68350c0
01-30 10:47:22.502 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttribPointer at 0xe6835da0
01-30 10:47:22.503 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttribIFormat at 0xe6836068
01-30 10:47:22.503 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttribI4uiv at 0xe6835e58
01-30 10:47:22.504 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttribI4ui at 0xe6835e50
01-30 10:47:22.505 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttribFormat at 0xe6836064
01-30 10:47:22.505 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib4f at 0xe6835d98
01-30 10:47:22.506 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib3fv at 0xe6835d94
01-30 10:47:22.506 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib2fv at 0xe6835d8c
01-30 10:47:22.507 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib2f at 0xe6835d88
01-30 10:47:22.507 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib1fv at 0xe6835d84
01-30 10:47:22.508 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glVertexAttrib1f at 0xe6835d80
01-30 10:47:22.509 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glUseProgramStagesEXT at 0xe68363e4
01-30 10:47:22.509 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glUseProgramStages at 0xe6835f70
01-30 10:47:22.510 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glUseProgram at 0xe6835d78
01-30 10:47:22.510 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glUnmapBuffer at 0xe6835ddc
01-30 10:47:22.511 21230 21230 E GAPID   : [gapii/cc/android/installer.cpp:232] Couldn't intercept function glUniformMatrix4x3fvNV at 0xe68364bc

```

**Comment by CatDroid** [comment]
p20 has the same log , and finally crash exception with 
```
01-30 11:02:29.399  3487  3598 E [HwRecSys]HwRecSysService: getIsRportDirectService is false!
01-30 11:02:29.445 13119 13173 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 13173 (RenderThread), pid 13119 (com.share)
01-30 11:02:29.471 13119 13179 E GLContextService: [run] Parent setup Begin
01-30 11:02:29.513 13180 13180 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-30 11:02:29.513 13180 13180 F DEBUG   : Build fingerprint: 'HUAWEI/EML-AL00/HWEML:9/HUAWEIEML-AL00/181C00R1:user/release-keys'
01-30 11:02:29.513 13180 13180 F DEBUG   : Revision: '0'
01-30 11:02:29.513 13180 13180 F DEBUG   : ABI: 'arm'
01-30 11:02:29.514 13180 13180 F DEBUG   : pid: 13119, tid: 13173, name: RenderThread  >>> com.share <<<
01-30 11:02:29.514 13180 13180 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
01-30 11:02:29.514 13180 13180 F DEBUG   : Cause: null pointer dereference
01-30 11:02:29.514 13180 13180 F DEBUG   :     r0  00000000  r1  60010000  r2  00006956  r3  80000000
01-30 11:02:29.514 13180 13180 F DEBUG   :     r4  da92d223  r5  c8a19fc0  r6  c83e3348  r7  c83e3230
01-30 11:02:29.514 13180 13180 F DEBUG   :     r8  c83e3348  r9  c83e370c  r10 e4a7e800  r11 c83e3720
01-30 11:02:29.514 13180 13180 F DEBUG   :     ip  f04eec08  sp  c83e3130  lr  caa81a19  pc  00000000

```

**Comment by CatDroid** [comment]
huawei p20 is kirin970 android 9.0 with GPU Trubo  and  GPU is  G72
huawei v10 is kirin970 android 8.0 with GPU is G72 yet

**Comment by AWoloszyn** [comment]
There were some known issue with Huawei devices in 1.2.1. We have just released 1.3.1 which may solve your problem.
https://github.com/google/gapid/releases/tag/v1.3.1

Please comment here if there is still an issue, and I can re-open.

**Comment by CatDroid** [comment]
yes it fixed , thanks
